### PR TITLE
Fixes #15344: API for node key management

### PR DESCRIPTION
**Pull request by fanf**
https://issues.rudder.io/issues/15344

### CODE REVIEW

**Comment by amousset** [comment]
Will also require a doc update.

File: webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/repository/ldap/LDAPEntityMapper.scala [hunk 1]
```diff
@@ -271,6 +271,23 @@ class LDAPEntityMapper(
...
 
+  def parseAgentInfo(nodeId: NodeId, inventoryEntry: LDAPEntry): IOResult[(List[AgentInfo], KeyStatus)] = {
+    val keys =  inventoryEntry.valuesFor(A_PKEYS).map(Some(_))
+    for {
+      agentsName     <- {
+                         val agents = inventoryEntry.valuesFor(A_AGENTS_NAME).toSeq.map(Some(_))
+                         ZIO.foreach(agents.zipAll(keys,None,None)) {
+                           case (Some(agent),key) => AgentInfoSerialisation.parseCompatNonJson(agent,key)
+                           case (None,key)        => (Err.MissingMandatory(s"There was a public key defined for Node ${nodeId.value},"+
+                                                             " without a releated agent defined, it should not happen")).fail
```
Comment (amousset) [inline webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/repository/ldap/LDAPEntityMapper.scala]: related

File: webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/repository/ldap/LDAPNodeRepository.scala [hunk 2]
```diff
@@ -87,6 +94,54 @@ class WoLDAPNodeRepository(
...
+    else {
+      for {
+        // check that in the case of a certificate, we are using a certificate for the node
+        _             <- agentKey match {
+                           case Some(Certificate(value)) => SecurityToken.checkCertificateForNode(nodeId, Certificate(value))
+                           case _                        => UIO.unit
+                         }
+        con           <- ldap
+        existingEntry <- con.get(acceptedDit.NODES.NODE.dn(nodeId.value), A_KEY_STATUS :: A_AGENTS_NAME :: Nil:_*).notOptional(
+                    s"Cannot update node with id ${nodeId.value} : there is no node with that id"
```
Comment (amousset) [inline webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/repository/ldap/LDAPNodeRepository.scala]: no space before colon

File: webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/repository/ldap/LDAPNodeRepository.scala [hunk 3]
```diff
@@ -87,6 +94,54 @@ class WoLDAPNodeRepository(
...
+                         )
+        agentsInfo    <- mapper.parseAgentInfo(nodeId, existingEntry).chainError(
+                           s"Error when geting agent key information from LDAP entry for node with id ${nodeId.value} . Entry: ${existingEntry}"
```
Comment (amousset) [inline webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/repository/ldap/LDAPNodeRepository.scala]: getting

**Comment by fanf** [comment]
PR updated with a new commit

**Comment by fanf** [comment]
OK, squash merging this PR

**Pull request merged**